lib/spack/spack/test/environment_modifications.py

```diff
@@ -158,75 +181,155 @@ def test_unset(env):
         os.environ["UNSET_ME"]
 
 
-@pytest.mark.not_on_windows("Not supported on Windows (yet)")
-def test_filter_system_paths(miscellaneous_paths):
+@pytest.mark.parametrize(
+    "miscellaneous_system_paths, expected",
+    [
+        # Windows Paths
+        (
+            [
+                "C:\\",
+                "C:\\Program Files",
+                "C:\\Program Files (x86)",
+                "C:\\Users",
+                "C:\\ProgramData",
+                "C:\\dev\\spack_window",
+            ],
+            (
+                ["C:\\dev\\spack_window"]
+                if sys.platform == "win32"
+                else [
+                    "C:\\",
+                    "C:\\Program Files",
+                    "C:\\Program Files (x86)",
+                    "C:\\Users",
+                    "C:\\ProgramData",
+                    "C:\\dev\\spack_window",
+                ]
```

Comment:
I think it would be best to exclude the windows parameterizations on non-Windows (and the other way around too).

Reply:
Agreed

Reply:
Done.